php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1174 @@
+      f.methodDeclaration(
+        b.zeroOrMore(MEMBER_MODIFIER()),
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        b.firstOf(
+          EOS(),
+          BLOCK())
+      ));
+  }
+
+  public FunctionDeclarationTree FUNCTION_DECLARATION() {
+    return b.<FunctionDeclarationTree>nonterminal(PHPLexicalGrammar.FUNCTION_DECLARATION).is(
+      f.functionDeclaration(
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        BLOCK()));
+  }
+  
+  public ParameterListTree PARAMETER_LIST() {
+    return b.<ParameterListTree>nonterminal(PHPLexicalGrammar.PARAMETER_LIST).is(
+      f.parameterList(
+        b.token(PHPPunctuator.LPARENTHESIS),
+        b.optional(
+          f.newTuple94(
+            PARAMETER(), 
+            b.zeroOrMore(
+              f.newTuple93(
+                b.token(PHPPunctuator.COMMA), 
+                PARAMETER())))),
+        b.token(PHPPunctuator.RPARENTHESIS)));
+  }
+  
+  public ParameterTree PARAMETER() {
+    return b.<ParameterTree>nonterminal(PHPLexicalGrammar.PARAMETER).is(
+      f.parameter(
+        b.optional(
+          b.firstOf(
+            b.token(PHPKeyword.ARRAY),
+            b.token(PHPKeyword.CALLABLE),
+            NAMESPACE_NAME())),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        b.optional(b.token(PHPPunctuator.ELIPSIS)),
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.optional(
+          f.newTuple92(
+            b.token(PHPPunctuator.EQU), 
+            STATIC_SCALAR())))
+      );      
+  }
+  
+  public SeparatedList<NamespaceNameTree> INTERFACE_LIST() {
+    return b.<SeparatedList<NamespaceNameTree>>nonterminal(PHPLexicalGrammar.INTERFACE_LIST).is(
+      f.interfaceList(
+        NAMESPACE_NAME(),
+        b.zeroOrMore(f.newTuple98(b.token(COMMA), NAMESPACE_NAME()))));
+  }
+  
+  public TraitUseStatementTree TRAIT_USE_STATEMENT() {
+    return b.<TraitUseStatementTree>nonterminal(PHPLexicalGrammar.TRAIT_USE_STATEMENT).is(
+      b.firstOf(
+        f.traitUseStatement(
+          b.token(PHPKeyword.USE),
+          INTERFACE_LIST(),
+          EOS()),
+        f.traitUseStatement(
+          b.token(PHPKeyword.USE),
+          INTERFACE_LIST(),
+          b.token(LCURLYBRACE),
+          b.zeroOrMore(
+            b.firstOf(
+              TRAIT_PRECEDENCE(),
+              TRAIT_ALIAS())),
+          b.token(RCURLYBRACE))));
+  }
+  
+  public TraitPrecedenceTree TRAIT_PRECEDENCE() {
+    return b.<TraitPrecedenceTree>nonterminal(PHPLexicalGrammar.TRAIT_PRECEDENCE).is(
+      f.traitPrecedence(
+        TRAIT_METHOD_REFERENCE_FULLY_QUALIFIED(),
+        b.token(PHPKeyword.INSTEADOF),
+        INTERFACE_LIST(),
+        EOS()));
+  }
+  
+  public TraitAliasTree TRAIT_ALIAS() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)